### PR TITLE
backport: refactor(wallet): simplify platform sync by removing PlatformSyncMode

### PR DESCRIPTION
**Pull request by lklimek**
## Summary

- Removes the `PlatformSyncMode` enum (Auto/ForceFull/TerminalOnly) and the 250-line `apply_recent_balance_changes()` method
- Replaces complex full/terminal sync logic with SDK-managed incremental sync via `sync_address_balances(provider, config, last_ts)`
- Removes `last_full_sync_balance` from `PlatformAddressInfo` and `last_terminal_block` from DB
- Simplifies `RefreshMode` to just All/CoreOnly/PlatformOnly
- Updates Platform SDK to rev `0fa82e6652097d17a700d8bcc006d6b2aa922c6e` which includes the incremental sync API

## Test plan

Manual test scenarios: [`docs/ai-design/2026-02-24-platform-sync-simplification/manual-test-scenarios.md`](docs/ai-design/2026-02-24-platform-sync-simplification/manual-test-scenarios.md)

- [ ] Fresh wallet sync (no prior sync state)
- [ ] Incremental sync (wallet already has balances)
- [ ] Refresh mode dropdown (All / Core Only / Platform Only)
- [ ] Balance after funding via asset lock / UTXOs
- [ ] Balance after withdrawal
- [ ] Locked wallet error path
- [ ] Network switching (Mainnet/Testnet/Regtest)
- [ ] Regtest empty balance tree
- [ ] No DAPI endpoints
- [ ] Database migration from old schema
- [ ] Pending refresh after transfer

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

### CODE REVIEW

**Comment by lklimek** [comment]
## 🔍 Audit Summary — PR #635

**Reviewed by:** Claude Code with a 3-agent team:
- `rust-quality-reviewer` (rust-developer, Opus) — code quality, correctness, idiomatic Rust
- `security-reviewer` (security-engineer, Opus) — fund safety, transaction integrity, race conditions
- `consistency-reviewer` (project-reviewer, Sonnet) — DB migrations, dead code, convention adherence

### Overall Assessment

**Risk: LOW-MEDIUM.** This PR is a **net improvement** — the UTXO two-phase select/remove pattern is a significant safety upgrade, the platform sync simplification removes ~250 lines of error-prone manual balance delta arithmetic, and the fee calculation is properly tested. The findings below are mostly pre-existing gaps surfaced by the refactoring, not new regressions.

### Findings

| # | Severity | Location | Description | Agents |
|---|----------|----------|-------------|--------|
| 1 | 🟡 Medium | `asset_lock_transaction.rs:430+` | **UTXOs removed before broadcast** — if broadcast fails after signing, UTXOs are removed from wallet tracking but never spent on-chain. Wallet shows incorrect balance until next UTXO reload. | Security, Rust |
| 2 | 🟡 Medium | `register_identity.rs:129+`, `top_up_identity.rs:145+` | **Store-after-broadcast race** — asset lock TX is broadcast before being stored in DB, leaving an SPV InstantSend race window. `fund_platform_address_from_wallet_utxos.rs` was correctly fixed to store-before-broadcast, but these two paths were not. | Security |
| 3 | 🟡 Medium | `initialization.rs:319,358` | **Fresh-install schema retains removed columns** — `last_terminal_block` and `last_full_sync_balance` are still in `create_tables()` DDL, but no code reads/writes them. No DB version bump to clean this up. | Consistency |
| 4 | 🟠 Low | `database/wallet.rs:1142` | **Column semantically repurposed** — `last_platform_sync_checkpoint` now stores SDK `new_sync_height`, different semantics from old "checkpoint". Rust variable renamed but SQL column not. | Consistency |
| 5 | 🟠 Low | `register_identity.rs:201,238`, `top_up_identity.rs:179,223` | **`.unwrap()` on RwLock in `FundWithUtxo` path** — adjacent `FundWithWallet` paths were correctly updated to `.map_err()`, but `FundWithUtxo` still uses `.unwrap()`. | Rust, Security |
| 6 | 🟠 Low | `database/wallet.rs:976` | **`_is_sync_operation` parameter does nothing** — underscore prefix suppresses warning but parameter is dead. All callers pass values that are ignored. | All 3 |
| 7 | 🟠 Low | `model/wallet/mod.rs:2002` | **`set_platform_address_info_from_sync` is trivial alias** — now identical to `set_platform_address_info`. Dead abstraction. | Consistency |
| 8 | 🟠 Low | `create_asset_lock.rs:20,63` | **Dead `_used_utxos` bindings** — UTXO cleanup moved into wallet method, but return values still captured as dead bindings. | Rust |
| 9 | ⚪ Nitpick | `fetch_platform_address_balances.rs:23` | `wallets.read().unwrap()` — inconsistent with `.map_err()` improvements elsewhere in this PR. | Rust, Security |
| 10 | ⚪ Nitpick | `fetch_platform_address_balances.rs:45-46` | Unnecessary `let mut provider = provider;` rebinding. | Rust |

### Pre-existing / Outside-diff Issues

Findings #1, #2, #3, #5, #8 affect files not in the GitHub diff (they're from backported commits already merged to `v1.0-dev`). These are noted for awareness but inline comments cannot be placed on them.

**Finding #1 detail**: The `remove_selected_utxos` call happens inside `asset_lock_transaction_from_private_key` (after signing, before broadcast). If broadcast subsequently fails, UTXOs are lost from tracking. Recommendation: either move `remove_selected_utxos` to after successful broadcast, add UTXO restoration on broadcast failure, or trigger a UTXO reload on error paths.

**Finding #2 detail**: `register_identity.rs` broadcasts at line 129, stores at line 135. `top_up_identity.rs` broadcasts at line 145, stores at line 148. The `fund_platform_address_from_wallet_utxos.rs` shows the correct pattern (store at line 87, broadcast at line 109, cleanup on failure at line 115). Apply the same pattern to the identity paths.

**Finding #3 detail**: `create_tables()` DDL in `initialization.rs` still includes `last_terminal_block INTEGER DEFAULT 0` (line 319) and `last_full_sync_balance INTEGER DEFAULT NULL` (line 358). Bump `DEFAULT_DB_VERSION` to 28, remove these from fresh-install DDL. Migration stubs for existing users are fine as-is.

### Positive Observations

- ✅ **Two-phase UTXO pattern** (`select_unspent_utxos_for` + `remove_selected_utxos`) eliminates the old fund-loss bug where signing failure permanently removed UTXOs
- ✅ **Overflow-safe arithmetic** — `checked_add` and `i64::try_from` replace raw `as` casts in financial calculations
- ✅ **Excellent test coverage** — `calculate_asset_lock_fee` has 7 targeted tests with "BUG TEST" comments explaining what each prevents
- ✅ **Clean PlatformSyncMode removal** — no orphan references anywhere, `RefreshMode` correctly collapsed to 3 variants
- ✅ **Store-before-broadcast** pattern correctly applied in `fund_platform_address_from_wallet_utxos.rs`
- ✅ **Comprehensive manual test scenarios** — 11 scenarios with edge-case checklist
- ✅ **SPV-aware reload logic** — `reload_utxos` correctly returns no-op in SPV mode

### Redundancy

3 agents reported 10 unique findings. Cross-agent overlap:
- Finding #6 (`_is_sync_operation`): flagged by all 3 agents (67% redundancy on this finding)
- Finding #5 (`.unwrap()` on locks): flagged by Rust + Security (33% redundancy)
- Overall redundancy ratio: ~20% (2 duplicated findings out of 10 unique)

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: src/database/wallet.rs [hunk 1]
```diff
@@ -982,19 +969,14 @@ impl Database {
...
-    /// can correctly apply any pending AddToCredits.
     pub fn set_platform_address_info(
         &self,
         seed_hash: &[u8; 32],
         address: &Address,
         balance: u64,
         nonce: u32,
         network: &Network,
-        is_sync_operation: bool,
+        _is_sync_operation: bool,
```
Comment (lklimek) [inline src/database/wallet.rs]: 🟠 **Low — Dead parameter** (Finding #6)

`_is_sync_operation` does nothing — the underscore prefix just suppresses the warning. The old two-path logic (sync vs non-sync) is gone, replaced by a single `INSERT ... ON CONFLICT DO UPDATE`. All callers still pass a boolean that is silently ignored.

```suggestion
        _is_sync_operation: bool, // TODO: remove this dead parameter and update all call sites
```

Alternatively, just remove the parameter entirely in this PR or a quick follow-up.

File: src/database/wallet.rs [hunk 2]
```diff
@@ -1167,49 +1123,31 @@ impl Database {
...
+                let sync_height: i64 = row.get(1)?;
+                Ok((last_sync as u64, sync_height as u64))
             },
         )
     }
 
-    /// Set the last platform full sync timestamp and checkpoint height for a wallet
-    /// Also resets last_terminal_block to 0 since a new full sync was performed
+    /// Set the platform sync timestamp and sync height for a wallet.
     pub fn set_platform_sync_info(
```
Comment (lklimek) [inline src/database/wallet.rs:1146]: 🟠 **Low — Semantic column name drift** (Finding #4)

This function now stores the SDK's `new_sync_height` into the `last_platform_sync_checkpoint` column. The Rust variable was renamed to `sync_height` but the SQL column still says "checkpoint" — different semantics.

Suggestion: At minimum, add a comment here noting the semantic change. Ideally, rename the column in a future DB migration.

File: src/model/wallet/mod.rs [hunk 3]
```diff
@@ -1964,97 +1960,52 @@ impl Wallet {
...
-        );
+        self.platform_address_info
+            .insert(address, PlatformAddressInfo { balance, nonce });
     }
 
-    /// Set platform address info from a sync operation.
-    /// Always updates `last_full_sync_balance` to the current balance, as this becomes
-    /// the baseline for pre-population in the next terminal sync.
+    /// Set platform address info from a sync operation (same as `set_platform_address_info`).
     pub fn set_platform_address_info_from_sync(
```
Comment (lklimek) [inline src/model/wallet/mod.rs]: 🟠 **Low — Dead abstraction** (Finding #7)

`set_platform_address_info_from_sync` is now identical to `set_platform_address_info` — it's a one-line delegate with the same behavior. The doc comment even says "same as `set_platform_address_info`".

Consider removing this method and updating the 2 call sites to use `set_platform_address_info` directly, or add `#[deprecated]` if cross-branch compatibility is needed.

File: src/backend_task/wallet/fetch_platform_address_balances.rs [hunk 4]
```diff
@@ -33,231 +27,97 @@ impl AppContext {
...
             match WalletAddressProvider::new(&wallet, self.network) {
-                Ok(provider) => provider,
+                Ok(provider) => provider.with_stored_state(&wallet, self.network, last_sync_height),
                 Err(_) if !wallet.is_open() => {
                     return Err("Wallet is locked. Please unlock it first to refresh.".to_string());
                 }
                 Err(e) => return Err(e),
             }
         };
+        let mut provider = provider;
```
Comment (lklimek) [inline src/backend_task/wallet/fetch_platform_address_balances.rs:45]: ⚪ **Nitpick — Unnecessary rebinding** (Finding #10)

```suggestion
        let mut provider = {
```

The provider is created immutable on line 35 then immediately rebound as mutable here. Combine into a single `let mut provider = { ... };` block.
